tools/fast/action_mining.go

```diff
@@ -4,9 +4,21 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/filecoin-project/go-filecoin/commands"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
+// MiningStatus runs the `mining status` command to get if mining
+func (f *Filecoin) MiningStatus(ctx context.Context) (bool, error) {
+	var out commands.MiningStatusResult
+
+	if err := f.RunCmdJSONWithStdin(ctx, nil, &out, "go-filecoin", "mining", "status"); err != nil {
+		return false, err
+	}
+
+	return out.Active, nil
+}
```

Comment:
This function should return the entire `commands.MiningStatusResult` structure.

Reply:
https://github.com/filecoin-project/go-filecoin/blob/1cab3122247e6a5c0f6a31227a649feb69f7aec2/tools/fast/action_miner.go#L18
https://github.com/filecoin-project/go-filecoin/blob/1cab3122247e6a5c0f6a31227a649feb69f7aec2/tools/fast/action_miner.go#L39
reference action_miner, it also return single value. and `commands.MiningStatusResult` have only one value.

Reply:
It could probably go either way at the moment, but the intent is that `miner status` will return additional information later. It's easier to return the status structure here now and now have to update everything down the road.

I would love to see `miner status` tell us the next proving period, proving set, etc.